joblib/test/test_numpy_pickle.py

```diff
@@ -598,6 +600,51 @@ def test_joblib_compression_formats():
                     os.remove(dump_filename)
 
 
+def _decompress_gzip(filename):
+    """Decompress a gzip file."""
+    with closing(gzip.GzipFile(filename + ".gz", "rb")) as fo:
+        buf = fo.read()
+
+    with open(filename, "wb") as fo:
+        fo.write(buf)
+
+
+def _decompress_zlib(filename):
+    """Decompress a zlib file."""
+    with open(filename + ".z", 'rb') as fo:
+        buf = zlib.decompress(fo.read())
+
+    with open(filename, 'wb') as fo:
+        fo.write(buf)
+
+
+def test_load_externally_decompressed_files():
+    # Test that BinaryZlibFile generates valid gzip and zlib compressed files.
+    obj = "a string to persist"
+    filename_raw = env['filename'] + str(random.randint(0, 1000))
+    compress_list = (('.z', _decompress_zlib),
+                     ('.gz', _decompress_gzip))
+
+    for extension, decompress_function in compress_list:
+        filename_compressed = filename_raw + extension
+        # Use automatic extension detection to compress with the right method.
+        numpy_pickle.dump(obj, filename_compressed)
+
+        # Decompress with the corresponding method
+        decompress_function(filename_raw)
+        nose.tools.assert_true(os.path.exists(filename_raw))
```

Comment:
Do we really need that? Hopefully the numpy_pickle.load below will fail with a meaningful exception.